apps/src/lib/kits/maker/boards/microBit/LightSensor.js

```diff
@@ -67,11 +82,39 @@ export default class LightSensor extends EventEmitter {
     this.board.mb.stopStreamingAnalogChannel(sensor_channels.lightSensor); // disable light sensor
   }
 
-  //TODO
-  getAveragedValue() {}
+  // Get the averaged value over the give range of ms, adjusted within specified range
+  // TODO: Handle if the range is outside of 50 (before first sample) or 3000 (longer than our buffer).
+  // Currently returns null if range is outside of acceptable values
+  getAveragedValue(range) {
+    if (range >= SAMPLE_RATE && range <= MAX_SENSOR_BUFFER) {
+      // Divide ms range by sample rate of sensor
+      let indicesRange = Math.ceil(range / SAMPLE_RATE);
+      let sum = 0;
+      for (let i = 0; i < indicesRange; i++) {
+        // bufferIndex points to the next spot to write, so first historical
+        // data value is at (bufferIndex - 1)
+        let index =
+          (this.buffer.length + (this.bufferIndex - (i + 1))) %
+          this.buffer.length;
+        sum += this.buffer[index];
+      }
```

Comment:
I found that the main assignment to `index` packed a few ideas into one line, and unexpectedly iterated backwards instead of forwards through the array (though for an average that's obviously okay).  Here's an attempt at unpacking things just a little bit:

```
const startIndex = this.bufferIndex - indicesRange;
const endIndex = this.bufferIndex - 1;

for (let index = startIndex; index <= endIndex; index++) {
  const actualIndex = (index + this.buffer.length) % this.buffer.length;
  sum += this.buffer[actualIndex];
}
```